packages/collector/Collector.php

```diff
@@ -0,0 +1,229 @@
+*/
+
+const COLLECTOR_DOWNLOAD_PATH   = '/wp-admin/?page=collector_download_package';
+const COLLECTOR_FINAL_ZIP       = '/tmp/collector-package.zip';
+
+define('COLLECTOR_PLAYGROUND_PACKAGE', ($_SERVER['SERVER_NAME'] === 'localhost')
+	? 'http://localhost:8081/index.js'
+	: 'https://unpkg.com/@wp-playground/client/index.js'
+);
+
+global $wp_version;
+
+define('COLLECTOR_WP_VERSION', $wp_version);
+define('COLLECTOR_PHP_VERSION', implode('.',sscanf(phpversion(), '%d.%d')));
+
+require __DIR__ . '/Collector_Content.php';
+require __DIR__ . '/Collector_Db.php';
+require __DIR__ . '/Collector_Helpers.php';
+require __DIR__ . '/Collector_Zip.php';
+
+add_action('admin_menu', 'collector_plugin_menu');
+add_action('plugins_loaded', 'collector_plugins_loaded');
+add_filter('plugin_install_action_links', 'collector_plugin_install_action_links', 10, 2);
+
+function collector_plugins_loaded()
+{
+    if(!current_user_can('manage_options'))
+    {
+        return;
+    }
+
+    if(urldecode($_SERVER['REQUEST_URI']) === COLLECTOR_DOWNLOAD_PATH)
+    {
+        collector_zip_collect();
+        collector_zip_send();
+        collector_zip_delete();
+        exit();
+    }
+}
+
+function collector_plugin_menu()
+{
+    add_submenu_page(
+        NULL,
+        'Collector',
+        'Collector',
+        'manage_options',
+        'collector_render_playground_page',
+        'collector_render_playground_page',
+        NULL
+    );
+}
+
+function collector_render_playground_page()
+{?>
+	<div id = "wp-playground-wrapper">
+		<div id = "wp-playground-toolbar">
+			NOW WORKING INSIDE WORDPRESS PLAYGROUND &nbsp; [<a href = "/wp-admin" id = "goBack">RETURN</a>]
+		</div>
+		<div id = "wp-playground-main-area">
+			<iframe id = "wp-playground"></iframe>
+		</div>
+	</div>
+	<script type = "text/javascript">
+		const frame  = document.getElementById('wp-playground');
+		const zipUrl = <?=json_encode(COLLECTOR_DOWNLOAD_PATH);?>;
+
+		const username   = <?=json_encode(wp_get_current_user()->user_login);?>;
+		const fakepass   = <?=json_encode(collector_get_fakepass());?>;
+		const pluginUrl  = new URLSearchParams(window.location.search).get('pluginUrl');
+        const pluginName = new URLSearchParams(window.location.search).get('pluginName');
+
+		(async () => {
+			const  { startPlaygroundWeb } = await import(<?=json_encode(COLLECTOR_PLAYGROUND_PACKAGE);?>);
+
+			const steps = [
+				{
+					step: 'writeFile',
+					path: '/data.zip',
+					data: {
+						'resource': 'url',
+						'url': zipUrl,
+					},
+				},
+				{
+					step: 'unzip',
+					zipPath: '/data.zip',
+					extractToPath: '/wordpress',
+				},
+				{
+					step: 'rm',
+					path: '/data.zip',
+				},
+				{
+					step: 'runSqlFile',
+					path: '/wordpress/schema/_Schema.sql',
+				},
+				{
+					step: 'rm',
+					path: '/wordpress/wp-content/mu-plugins/1-show-admin-credentials-on-wp-login.php',
+				},
+			];
+
+			if(pluginUrl && pluginName)
+			{
+				steps.push(...[
```

Comment:
Neat trick! At the same time, it took me a good few seconds to go from "wait, what?", "oh, push accepts multiple arguments so this code is pushing many items, so effectively concatenating two arrays". While this code is more verbose, it could save future readers the same mental process:

```js
steps.push({
     step: 'writeFile',
     path: '/plugin.zip',
     // ...
});

steps.push({
     step: 'unzip',
     zipPath: '/plugin.zip',
     // ...
});

// ...etc
```

Reply:
Ah, I just realized this would work in the same way by just removing the `...[` part:

```js

steps.push(
    {
          step: 'writeFile',
          path: '/plugin.zip',
          // ...
    },
    {
          step: 'unzip',
          zipPath: '/plugin.zip',
          // ...
     }
)
```

Reply:
This notation was the result of a bunch of block-level refactoring. I'll fix that.

Reply:
@adamziel 

https://github.com/WordPress/playground-tools/pull/124/files#diff-72b9ba31c538d25a3db28fcc6b6f58ca9b1fca1f66bab3947d565eaf85bbff07R117